.github/workflows/build.yml

```diff
@@ -0,0 +1,32 @@
+on:
+  pull_request:
+    paths-ignore:
+      - 'docs/**'
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.gitignore'
+      - '.gitattributes'
+      - 'infection.json.dist'
+      - 'psalm.xml'
+
+  push:
+    paths-ignore:
+      - 'docs/**'
+      - 'README.md'
+      - 'CHANGELOG.md'
+      - '.gitignore'
+      - '.gitattributes'
+      - 'infection.json.dist'
+      - 'psalm.xml'
+
+name: build
+
+jobs:
+  phpunit:
+    uses: yiisoft/actions/.github/workflows/phpunit.yml@master
+    with:
+      extensions: mbstring, intl
+      os: >-
+        ['ubuntu-latest', 'windows-latest']
+      php: >-
+        ['7.4', '8.0', '8.1', '8.2', '8.3']
```

Comment:
There will be legacy support, only that for the tests, they will be executed from PHP 7.4, since you would have to install PHP 4 with the patches, which is tedious to maintain with the new versions of PHP, but nothing prevents you from installing with the legacy versions, the code remains intact.